core/ErrorFlusher.h

```diff
@@ -10,15 +10,11 @@ class FileSystem;
 namespace core {
 
 class ErrorFlusher {
-private:
-    std::vector<AutocorrectSuggestion> autocorrects;
-    bool printedAtLeastOneError{false};
-
 public:
-    ErrorFlusher() = default;
-    void flushErrors(spdlog::logger &logger, std::vector<std::unique_ptr<ErrorQueueMessage>> error);
-    void flushErrorCount(spdlog::logger &logger, int count);
-    void flushAutocorrects(const GlobalState &gs, FileSystem &fs);
+    virtual void flushErrors(spdlog::logger &logger, std::vector<std::unique_ptr<ErrorQueueMessage>> error) = 0;
+    virtual void flushErrorCount(spdlog::logger &logger, int count) = 0;
+    virtual void flushAutocorrects(const GlobalState &gs, FileSystem &fs) = 0;
+    virtual ~ErrorFlusher(){};
```

Comment:
You can just make this the default.
```suggestion
    virtual ~ErrorFlusher() = default;
```